src/types.rs

```diff
@@ -902,11 +1028,81 @@ impl Term {
     /// If the constraints are not satisfiable, `None` is retuned. Constraints are in the form of
     /// patterns, where substitutions are only considered for variables in the first term of each
     /// pair.
```

Comment:
What do you think about expanding on the documentation of both `pmatches` and `unify` to explain (or at least link to) the difference between matching and unification? We could also give the same three examples across the two functions: one which neither matches nor unifies, one which matches and unifies, and one which unifies but doesn't match. How does that sound to you?